backend/internal/pkg/repositories/parkingspot/postgres.go

```diff
@@ -0,0 +1,141 @@
+package parkingspot
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/dbmodels"
+	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/models"
+	"github.com/aarondl/opt/omit"
+	"github.com/google/uuid"
+	"github.com/stephenafamo/bob"
+	"github.com/stephenafamo/bob/dialect/psql"
+	"github.com/stephenafamo/bob/dialect/psql/dm"
+	"github.com/stephenafamo/bob/dialect/psql/sm"
+	"github.com/stephenafamo/scan"
+)
+
+type PostgresRepository struct {
+	db bob.DB
+}
+
+func NewPostgres(db bob.DB) *PostgresRepository {
+	return &PostgresRepository{
+		db: db,
+	}
+}
+
+func (p *PostgresRepository) Create(ctx context.Context, userID64 int64, spot *models.ParkingSpotCreationInput) (int64, Entry, error) {
```

Comment:
Why are we passing an int64 just to convert them to int32? is It because there is discrepancy between postgres uuid type and go uuid type?